crates/turbopack-node/src/pool.rs

```diff
@@ -55,11 +60,53 @@ impl Drop for RunningNodeJsPoolProcess {
 
 const CONNECT_TIMEOUT: Duration = Duration::from_secs(30);
 
+type SharedOutputSet = Arc<Mutex<IndexSet<(Arc<[u8]>, u32)>>>;
+
+/// Pipes the `stream` from `final_stream`, but uses `shared` to deduplicate
+/// lines that has beem emitted by other `handle_output_stream` instances with
+/// the same `shared` before.
+async fn handle_output_stream(
+    stream: impl AsyncRead + Unpin,
+    shared: SharedOutputSet,
+    mut final_stream: impl AsyncWrite + Unpin,
+) {
+    let mut buffered = BufReader::new(stream);
+    let mut own_output = HashMap::<Arc<[u8]>, u32>::new();
+    let mut buffer = Vec::new();
+    loop {
+        match buffered.read_until(b'\n', &mut buffer).await {
```

Comment:
Do we expect non-UTF-8 sequences here? Otherwise let's just use `buffered.lines()`.

Reply:
I had this before, but lines is weird since it doesn't include the line terminator, which requires adding it manually before piping which requires extending the Vec by one byte, which potentially copies the data... This read_until is better